File: frontend/lib/models/review_model.dart
class ReviewModel  {
  final String id;

  final String videoId;
  
  final String comment;
  
  final double rating;
  
  final DateTime timestamp;
  
  ReviewModel({
    required this.id,
    required this.videoId,
    required this.comment,
    required this.rating,
    required this.timestamp,
  });
}

File: backend/models/review.model.js
import mongoose from "mongoose";

const mongoose = require("mongoose");

const reviewSchema = new mongoose.Schema(
  {
    id: {
      type: String,
      required: true,
      unique: true,
    },
    videoId: {
      type: String,
      required: true,
    },
    comment: {
      type: String,
      required: true,
    },
    rating: {
      type: Number,
      required: true,
      min: 0,
      max: 5,
    },
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
  },
  { timestamps: true }
);

const Review = mongoose.models('Review', reviewSchema);

export default Review